core/transport/src/main/proto/grpc/block_master.proto

```diff
@@ -223,14 +226,22 @@ message CommitBlockInUfsPRequest {
 message CommitBlockInUfsPOptions {}
 message CommitBlockInUfsPResponse {}
 
-message GetWorkerIdPOptions {}
+message GetWorkerIdPOptions {
+  optional bool extendedRegisterInfo = 1; // Identify whether there is preRegister information
+}
 message GetWorkerIdPRequest {
   /** the worker network address */
   optional grpc.WorkerNetAddress workerNetAddress = 1;
+  /** for preRegister information **/
   optional GetWorkerIdPOptions options = 2;
+  optional string clusterId = 3;
+  optional bool hasBlockInTier = 4;
```

Comment:
instead of a boolean, pass the current block count in the message so later the master can add some extra logic on that

Reply:
done